src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -10,21 +11,43 @@
  */
 public class SortCommandParser implements Parser<SortCommand> {
 
+    private static final Logger logger = LogsCenter.getLogger(AddressBookParser.class);
+
     /**
      * Parses the given {@code String} of arguments in the context of the SortCommand
      * and returns a SortCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public SortCommand parse(String args) throws ParseException {
+
         String trimmedArgs = args.trim();
+        String[] keywords = trimmedArgs.split("\\s+");
 
-        if (trimmedArgs.isEmpty()
-                || !(trimmedArgs.equals("name")
-                || trimmedArgs.equals("deadline"))) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        if (keywords.length != 2) {
+            throw new ParseException(SortCommand.MESSAGE_USAGE);
         }
 
-        return new SortCommand(trimmedArgs);
+        String sortBy = keywords[0].toLowerCase();
+        String order = keywords[1].toLowerCase();
+
+        logger.fine("Keyword for sort: " + sortBy + ", Keyword for order: " + order);
+
+        if (!(sortBy.equals("name") || sortBy.equals("deadline"))) {
+            throw new ParseException(SortCommand.MESSAGE_INVALID_KEYWORD);
+        }
+
+        boolean isAscending;
+        if (order.equalsIgnoreCase("ascending")) {
```

Comment:
```suggestion
        if (order.equals("ascending")) {
```
Since you're making it lower case earlier already